java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/CfDataSourceEnvironmentPostProcessor.java

```diff
@@ -95,6 +108,42 @@ public void postProcessEnvironment(ConfigurableEnvironment environment,
 				if (driverClassName != null) {
 					properties.put("spring.datasource.driver-class-name", driverClassName);
 				}
+				
+				/* R2DBC processing
+				 * Split query param options and URL into two string
+				 * and move options to spring.r2dbc.properties.<option>
+				 */
+				
+				String[] splitJDBCUrl = cfJdbcService.getJdbcUrl().split("\\?");
```

Comment:
I'd need to look at it in more detail but... why did you put this code, adding *rdbc* properties, in the `boot` module module rather than in the `jdbc` module?

Is it because you wanted your code to be *SQL agnostic (compatible with postgres and mysql) ?

Reply:
Yes, this was intended to be as agnostic as possible.  It also followed a similar model to spring-cloud-bindings where the R2DBC properties were derived from the JDBC properties (and was all placed into a single processor).  Example:  https://github.com/spring-cloud/spring-cloud-bindings/blob/main/spring-cloud-bindings/src/main/java/org/springframework/cloud/bindings/boot/PostgreSqlBindingsPropertiesProcessor.java